packages/repco-webcomponent/src/repco-post-element/repco-post-element.ts

```diff
@@ -0,0 +1,155 @@
+      box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1);
+      border-color: #cbd5e0;
+    }
+    :host([theme='dark']) .card {
+      background-color: var(--dark-card-background-color, #383838);
+      color: var(--dark-card-color, #f7fafc);
+      box-shadow: var(--dark-card-shadow, 0px 4px 6px rgba(0, 0, 0, 0.1));
+      border-color: var(--dark-card-border-color, #cbd5e0);
+    }
+    :host([theme='dark']) .card:hover {
+      background-color: var(--dark-card-hover-background-color, #1c1c1c);
+      box-shadow: var(--dark-card-hover-shadow, 0px 4px 6px rgba(0, 0, 0, 0.1));
+      border-color: var(--dark-card-hover-border-color, #cbd5e0);
+    }
+  `
+  @property({ type: String, reflect: true })
+  theme = 'light'
+
+  @property({ type: String })
+  layout = 'column'
+
+  @property({ type: Number })
+  count = 10
+
+  @property({ type: String })
+  _defaultThumbnail = ''
+
+  @property()
+  endpoint = 'https://node1.repco.openaudiosearch.org/graphql'
+
+  @property({ type: String })
+  query = `
+  query {
+    contentItems {
+      nodes {
+        title
+        content
+        uid
+        mediaAssets {
+          nodes {
+            mediaType
+            file {
+              contentUrl
+            }
+          }
+        }
+      }
+    }
+  }
+`
+  @state()
+  private _state: { posts: PostType[] } = { posts: [] }
+
+  private _repcoFetchClient: any
+
+  override async firstUpdated() {
+    this.fetchData()
+  }
+
+  override render() {
+    return html` <repco-fetch-api
+        .endpoint="${this.endpoint}"
+        .query="${this.query}"
+      >
+      </repco-fetch-api>
+      <div
+        style="display: flex; flex-direction: ${this.layout === 'horizontal'
+          ? 'row'
+          : 'column'};"
+      >
+        ${this._state.posts.slice(0, this.count).map((post) => {
+          let thumbnail = this._defaultThumbnail
+          post.mediaAssets.nodes.map((asset: MediaAssetType) => {
+            if (asset.mediaType == 'image') {
+              thumbnail = asset.file.contentUrl
+            }
+          })
+
+          const header = post.title
+          const subheader = post.uid
+
+          const body = `${this.trimContent(post.content)}...`
+          const footer = `source: ${this.endpoint}`
+          const endpointBase = this.endpoint.substring(
+            0,
+            this.endpoint.lastIndexOf('/'),
+          )
+          const link = `${endpointBase}/items/${post.uid}`
+
+          return html`
+            <repco-post-card
+              class="card"
+              .thumbnail="${thumbnail}"
+              .header="${header}"
+              .subheader="${subheader}"
+              .body="${body}"
+              .footer="${footer}"
+              @click=${() => this.cardClick(link)}
+            ></repco-post-card>
+          `
+        })}
+      </div>`
+  }
+
+  private async fetchData() {
+    if (this.shadowRoot) {
+      this._repcoFetchClient =
+        this.shadowRoot.querySelector('repco-fetch-client')
```

Comment:
There it queries for the repco-fetch-client componente check the dom. When the repco-fetch-client is rendered only then he can get the data from it because only then the query was executed. So it is the name of the fetch-client-component

Reply:
but that repco-fetch-client component doesn't exist as far as I can see? shouldn't that be `repco-fetch-api` then?

Reply:
yes this is why I wonder how it works ....